backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/database/repos/CardRepository.kt

```diff
@@ -21,9 +21,18 @@ object CardRepository {
         return if (query == null) null else CardEntity.wrapRow(query)
     }
 
-    fun insert(cardInfoHash: ByteArray, totpSecret: ByteArray?, expirationDay: Long?, regionId: Int, issuerId: Int, codeType: CodeType) =
+    fun findByHashModelAndActivationSecret(project: String, hashModel: ByteArray, activationSecret: ByteArray): CardEntity? {
+        val query = (Projects innerJoin Regions innerJoin Cards)
+            .slice(Cards.columns)
+            .select { Projects.project eq project and (Cards.cardInfoHash eq hashModel) and (Cards.activationSecret eq activationSecret) }
+            .singleOrNull()
+        return if (query == null) null else CardEntity.wrapRow(query)
+    }
```

Comment:
```suggestion
    fun findByHashAndActivationSecret(project: String, cardInfoHash: ByteArray, activationSecret: ByteArray): CardEntity? {
        val query = (Projects innerJoin Regions innerJoin Cards)
            .slice(Cards.columns)
            .select { Projects.project eq project and (Cards.cardInfoHash eq cardInfoHash) and (Cards.activationSecret eq activationSecret) }
            .singleOrNull()
        return if (query == null) null else CardEntity.wrapRow(query)
    }
```

(If you want you can also adjust the name of the function above to `findByHash` or `findByCardInfoHash`.